File: c_plus_plus/tests/singleroundtests/orientationtest.cpp
#include "orientationtest.h"
#include <QTest>
#include "plane.h"



void OrientationTest::initTestCase()
{
    qDebug("OrientationTest starts ..");
}

void OrientationTest::Orientation_Values() {
    QVERIFY(0 == (int)Plane::Orientation::NorthSouth);
    QVERIFY(1 == (int)Plane::Orientation::SouthNorth);
    QVERIFY(2 == (int)Plane::Orientation::WestEast);
    QVERIFY(3 == (int)Plane::Orientation::EastWest);
}


void OrientationTest::cleanupTestCase()
{
    qDebug("OrientationTest ends ..");
}

File: kotlin/PlanesAndroid/app/src/test/java/com/planes/singleround/test/OrientationTest.kt
package com.planes.singleround.test.com.planes.singleround.test
import com.google.common.truth.Truth
import com.planes.single_player_engine.Orientation
import org.junit.Test

class OrientationTest {
    @Test
    fun Orientation_Values() {
        Truth.assertThat(Orientation.values()[0] === Orientation.NorthSouth).isTrue()
        Truth.assertThat(Orientation.values()[1] === Orientation.SouthNorth).isTrue()
        Truth.assertThat(Orientation.values()[2] === Orientation.WestEast).isTrue()
        Truth.assertThat(Orientation.values()[3] === Orientation.EastWest).isTrue()
    }
}